### PR TITLE
Added CopyJSMojo for copying js files

### PR DESCRIPTION
**Pull request by adrianfish**

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -3,7 +3,7 @@
     <name>Sakai Maven Plugin</name>
     <groupId>org.sakaiproject.maven</groupId>
     <artifactId>plugins</artifactId>
-    <version>1.4.6-SNAPSHOT</version>
+    <version>1.4.7-SNAPSHOT</version>
```
Comment (jonespm) [inline pom.xml]: I don't think we ever actually released 1.4.6, the version bump should only be happening via the maven release plugin. This process is documented in the README.

File: sakai-plugin/src/main/java/org/sakaiproject/maven/plugin/component/CopyJSMojo.java [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+          } catch (FileAlreadyExistsException e) {
+            if (!Files.isDirectory(currentTargetDir)) throw e;
+          }
+          return FileVisitResult.CONTINUE;
+        }
+
+        @Override
+        public FileVisitResult visitFile(Path file, BasicFileAttributes attrs) throws IOException {
+
+          if (file.toString().endsWith(".js")) {
```
Comment (jonespm) [inline sakai-plugin/src/main/java/org/sakaiproject/maven/plugin/component/CopyJSMojo.java:52]: Do you think we'd also have to be concerned with .ts files?

File: sakai-plugin/src/main/java/org/sakaiproject/maven/plugin/component/CopyJSMojo.java [hunk 3]
```diff
@@ -0,0 +1,96 @@
...
+
+  private void transform(Path file, Path targetFile) {
+
+    try {
+      try (BufferedReader reader = Files.newBufferedReader(file);
+              BufferedWriter writer = Files.newBufferedWriter(targetFile, StandardOpenOption.CREATE)) {
+
+        reader.lines().forEach(line -> {
+
+          try {
```
Comment (jonespm) [inline sakai-plugin/src/main/java/org/sakaiproject/maven/plugin/component/CopyJSMojo.java:75]: Can you add some comments around this process or document this method. I'm not sure what this transform is doing here just by looking at the code quickly. If you're trying to do some kind of URL parsing library that would simplify some of this?

**Review by jonespm — CHANGES_REQUESTED** [review]

File: sakai-plugin/src/main/java/org/sakaiproject/maven/plugin/component/CopyJSMojo.java (hunk 2 above)
Comment (adrianfish) [reply sakai-plugin/src/main/java/org/sakaiproject/maven/plugin/component/CopyJSMojo.java:52]: Don't think so. We've no plans around typescript at the moment. I mean, we could add it for future proofing sake, but I don't think it's necessary.

File: sakai-plugin/src/main/java/org/sakaiproject/maven/plugin/component/CopyJSMojo.java (hunk 3 above)
Comment (adrianfish) [reply sakai-plugin/src/main/java/org/sakaiproject/maven/plugin/component/CopyJSMojo.java:75]: Yep, will do!

File: sakai-plugin/src/main/java/org/sakaiproject/maven/plugin/component/CopyJSMojo.java (hunk 2 above)
Comment (jonespm) [reply sakai-plugin/src/main/java/org/sakaiproject/maven/plugin/component/CopyJSMojo.java:52]: Okay just checking, I guess it's not a big deal if we don't need it now.

**Review by jonespm — APPROVED** [review]

**Pull request merged**